src/Search/Api/Elasticsearch/Command/BuildIndexCommand.php

```diff
@@ -59,15 +63,16 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $create = null;
 
         // Try removing old one.
-        try {
-            $delete = $this->client->deleteIndex();
-        } catch (Throwable $e) {
-            $logger->debug($e->getMessage(), $e->getTrace());
+        if ($toDelete) {
+            try {
+                $delete = $this->client->deleteIndex();
+            } catch (Throwable $e) {
+                $logger->debug($e->getMessage(), $e->getTrace());
```

Comment:
Standard way to log an exception during an important setup operation like this should be:
```
$logger->error("Cannot delete index", ['exception' => $e]);
```
Exceptions in the context can be dumped with the right level of verbosity (including stack traces) by log handlers.